src/main/java/com/linkedin/kmf/services/KafkaMetricsReporterService.java

```diff
@@ -0,0 +1,144 @@
+ * file except in compliance with the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.linkedin.kmf.services;
+
+import static com.linkedin.kmf.common.Utils.getMBeanAttributeValues;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.linkedin.kmf.common.MbeanAttributeValue;
+import com.linkedin.kmf.common.Utils;
+import com.linkedin.kmf.services.configs.KafkaMetricsReporterServiceConfig;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Created by IntelliJ IDEA.
+ * User: hackerwin7
+ * Date: 2018/01/18
+ * Time: 11:30 AM
+ * Desc:
+ */
+public class KafkaMetricsReporterService implements Service {
+
+  private static final Logger LOG = LoggerFactory.getLogger(KafkaMetricsReporterService.class);
+
+  private final String _name;
+  private final List<String> _metricsNames;
+  private final int _reportIntervalSec;
+  private final ScheduledExecutorService _executor;
+
+  private KafkaProducer<String, String> _producer;
+  private final String _producerId;
+  private final String _brokerList;
+  private final String _zkConnect;
+  private final String _topic;
+
+  private final ObjectMapper _parser = new ObjectMapper();
+
+  public KafkaMetricsReporterService(Map<String, Object> props, String name) throws Exception {
+    _name = name;
+    KafkaMetricsReporterServiceConfig config = new KafkaMetricsReporterServiceConfig(props);
+    _metricsNames = config.getList(KafkaMetricsReporterServiceConfig.REPORT_METRICS_CONFIG);
+    _reportIntervalSec = config.getInt(KafkaMetricsReporterServiceConfig.REPORT_INTERVAL_SEC_CONFIG);
+    _executor = Executors.newSingleThreadScheduledExecutor();
+
+    _brokerList = config.getString(KafkaMetricsReporterServiceConfig.BOOTSTRAP_SERVERS_CONFIG);
+    _producerId = config.getString(KafkaMetricsReporterServiceConfig.PRODUCER_ID_CONFIG);
+    initializeProducer();
+
+    _zkConnect = config.getString(KafkaMetricsReporterServiceConfig.ZOOKEEPER_CONNECT_CONFIG);
+    _topic = config.getString(KafkaMetricsReporterServiceConfig.TOPIC_CONFIG);
+    Utils.createTopicIfNotExists(_zkConnect, _topic, 1, 1, new Properties()); // fixed partition and factor
```

Comment:
If broker number is more than 1, we probably want to set replication factor to be more than 1, right?

Reply:
Maybe I should put out the `replicationFactor` as a configuration parameter, and `partitionCount` is 
fixed `1` ? either, put out both as configuration ?

Reply:
It think it is reasonable to have replicationFactor config for KafkaMetricsReporterService. I suppose the throughput of the metrics messages should be low. So it is probably OK to fix the partition number to 1.